media/editor/dataInspectorProperties.tsx

```diff
@@ -66,70 +66,44 @@ export const inspectableTypes: readonly IInspectableType[] = [
 	{ label: "int64", minBytes: 8, convert: (dv, le) => dv.getBigInt64(0, le).toString() },
 	{ label: "uint64", minBytes: 8, convert: (dv, le) => dv.getBigUint64(0, le).toString() },
 
-	{ label: "float16", minBytes: 2, convert: (dv, le) => getFloat16(5, 10)(dv.buffer, le).toString() },
-	{ label: "bfloat16", minBytes: 2, convert: (dv, le) => getFloat16(8, 7)(dv.buffer, le).toString() },
-
-	{ label: "float32", minBytes: 4, convert: (dv, le) => dv.getFloat32(0, le).toString() },
-	{ label: "float64", minBytes: 8, convert: (dv, le) => dv.getFloat64(0, le).toString() },
-
-	{
-		label: "UTF-8",
-		minBytes: 1,
-		convert: dv => {
-			const utf8 = new TextDecoder("utf-8").decode(dv.buffer);
-			for (const char of utf8) return char;
-			return utf8;
-		},
-	},
 	{
-		label: "UTF-16",
+		label: "float16",
 		minBytes: 2,
-		convert: (dv, le) => {
-			const utf16 = new TextDecoder(le ? "utf-16le" : "utf-16be").decode(dv.buffer);
-			for (const char of utf16) return char;
-			return utf16;
-		},
+		convert: (dv, le) => getFloat16(5, 10)(dv.buffer, le).toString(),
 	},
 	{
-		label: "GB18030",
+		label: "bfloat16",
 		minBytes: 2,
-		convert: dv => {
-			// the valid encoding for TextDecoder is list on
-			// https://developer.mozilla.org/en-US/docs/Web/API/Encoding_API/Encodings
-			// GBK Character Set is an extension of GB2312. GB18030 is an extension of GBK.
-			// So choose GB18030 as the encoding here.
-			// see also http://herongyang.com/GB2312/Introduction-GB2312-GBK-GB18030.html
-			// and https://www.ibm.com/docs/en/aix/7.1?topic=sets-gb18030
-			const utf8 = new TextDecoder("gb18030").decode(dv.buffer);
-			for (const char of utf8) return char;
-			return utf8;
-		},
+		convert: (dv, le) => getFloat16(8, 7)(dv.buffer, le).toString(),
 	},
-	{
-		label: "BIG5",
-		minBytes: 2,
-		convert: dv => {
-			const utf8 = new TextDecoder("big5").decode(dv.buffer);
-			for (const char of utf8) return char;
-			return utf8;
-		},
-	},
-	{
-		label: "ISO-2022-KR",
-		minBytes: 2,
-		convert: dv => {
-			const utf8 = new TextDecoder("iso-2022-kr").decode(dv.buffer);
-			for (const char of utf8) return char;
-			return utf8;
-		},
-	},
-	{
-		label: "SHIFT-JIS",
-		minBytes: 2,
+
+	{ label: "float32", minBytes: 4, convert: (dv, le) => dv.getFloat32(0, le).toString() },
+	{ label: "float64", minBytes: 8, convert: (dv, le) => dv.getFloat64(0, le).toString() },
+];
+
+const addTextDecoder = (encoding: string, minBytes: number) => {
+	try {
+		new TextDecoder(encoding); // throws if encoding is now supported
+	} catch {
+		return;
+	}
+
+	inspectTypesBuilder.push({
+		label: encoding.toUpperCase(),
+		minBytes,
 		convert: dv => {
-			const utf8 = new TextDecoder("shift-jis").decode(dv.buffer);
+			const utf8 = new TextDecoder(encoding).decode(dv.buffer);
 			for (const char of utf8) return char;
 			return utf8;
 		},
-	},
-];
+	});
+};
+
+addTextDecoder("utf-8", 1);
```

Comment:
This was a separate issue caused by a PR that introduced `iso-2022-kr` encoding which is not supported on my machine and caused the hex editor to crash